components/test_raftstore/src/transport_simulate.rs

```diff
@@ -720,10 +731,10 @@ impl Filter for LeadingDuplicatedSnapshotFilter {
     }
 
     fn after(&self, res: Result<()>) -> Result<()> {
-        let dropped = self
-            .dropped
-            .compare_and_swap(true, false, Ordering::Relaxed);
-        if res.is_err() && dropped {
+        let dropped =
+            self.dropped
+                .compare_exchange(true, false, Ordering::Relaxed, Ordering::Relaxed);
+        if res.is_err() && dropped.is_ok() {
```

Comment:
I think moving `is_ok()` upward would be semantically better.

```suggestion
        let dropped =
            self.dropped
                .compare_exchange(true, false, Ordering::Relaxed, Ordering::Relaxed)
                .is_ok();
        if res.is_err() && dropped {
```